### PR TITLE
chore(Conditionbuilder): typescript migration

### LINKED ISSUE #5670: Add Typescript types to ConditionBuilder
As we add Typescript support, want to call out a few helpful tips from the Carbon TypeScript guide that is largely relevant to our library as well.

### Changing the extension via git

When changing the file extension — **do not simply rename the file**. **Instead, change the extension via `git`.**

For example,

```
git mv packages/ibm-products/src/components/ComponentName/ComponentName.js packages/ibm-products/src/components/ComponentName/ComponentName.tsx
```

### Do not add types to internal components or functions
- Leave internals as `.js`
- Add a [jsdoc type annotation](https://www.typescriptlang.org/docs/handbook/intro-to-js-ts.html#providing-type-hints-in-js-via-jsdoc) of `/** @type any */`
- [Enable errors](https://www.typescriptlang.org/docs/handbook/intro-to-js-ts.html#ts-check) by adding `// @ts-check` to the first line in the file.

### Testing changes
A few different options for testing changes —

- At the bottom of a file, write a dummy component that uses the component that you’re converting to ensure that you can still pass all appropriate props.
- Take a storybook example for a component, copy and paste it into the bottom of the `.tsx` file and validate whether or not it accepts the props as you’ve defined them.

```[tasklist]
### Tasks
- [ ] Change extension to `.tsx` **via git** (**do not simply rename**, see above)
- [ ] Copy the PropTypes definitions above the component definition
- [ ] Retool the PropTypes to be a TypeScript interface
- [ ] Fix errors as they appear
- [ ] Do NOT add types to internal components or functions that are not exported as part of the public API
- [ ] Test your changes (see above)
```

### Resources
Carbon’s guide on [adding component types](https://github.com/carbon-design-system/carbon/blob/main/docs/guides/adding-component-types.md)

### PR DESCRIPTION
**Pull request by amal-k-joy**
Closes #5670 

Typescript migration for condition builder

#### What did you change?
```

1. packages/ibm-products/src/components/ConditionBuilder/ConditionBuilder.types.ts
2. packages/ibm-products/src/components/ConditionBuilder/ConditionBlock/ConditionBlock.tsx
3. packages/ibm-products/src/components/ConditionBuilder/ConditionBuilder.tsx
4. packages/ibm-products/src/components/ConditionBuilder/ConditionBuilderActions/ConditionBuilderActions.tsx
5. packages/ibm-products/src/components/ConditionBuilder/ConditionBuilderAdd/ConditionBuilderAdd.tsx
6. packages/ibm-products/src/components/ConditionBuilder/ConditionBuilderButton/ConditionBuilderButton.tsx
7. packages/ibm-products/src/components/ConditionBuilder/ConditionBuilderConnector/ConditionConnector.tsx
8. packages/ibm-products/src/components/ConditionBuilder/ConditionBuilderContent/ConditionBuilderContent.tsx
9. packages/ibm-products/src/components/ConditionBuilder/ConditionBuilderContext/ConditionBuilderProvider.tsx
10. packages/ibm-products/src/components/ConditionBuilder/ConditionBuilderItem/ConditionBuilderItem.tsx
11. packages/ibm-products/src/components/ConditionBuilder/ConditionBuilderItem/ConditionBuilderItemDate/ConditionBuilderItemDate.tsx
12. packages/ibm-products/src/components/ConditionBuilder/ConditionBuilderItem/ConditionBuilderItemNumber/ConditionBuilderItemNumber.tsx
13. packages/ibm-products/src/components/ConditionBuilder/ConditionBuilderItem/ConditionBuilderItemOption/ItemOption.tsx
14. packages/ibm-products/src/components/ConditionBuilder/ConditionBuilderItem/ConditionBuilderItemOption/ItemOptionForValueField.tsx
15. packages/ibm-products/src/components/ConditionBuilder/ConditionBuilderItem/ConditionBuilderItemText/ConditionBuilderItemText.tsx
16. packages/ibm-products/src/components/ConditionBuilder/ConditionBuilderItem/ConditionBuilderItemTime/ConditionBuilderItemTime.tsx
17. packages/ibm-products/src/components/ConditionBuilder/ConditionGroupBuilder/ConditionGroupBuilder.tsx
18. packages/ibm-products/src/components/ConditionBuilder/ConditionPreview/ConditionPreview.tsx

```
#### How did you test and verify your work?
TS error and storybook

### CODE REVIEW

**Review by AlexanderMelox — CHANGES_REQUESTED** [review]

**Review by davidmenendez — CHANGES_REQUESTED** [review]
agreed with @AlexanderMelox

File: packages/ibm-products/src/components/ConditionBuilder/ConditionBlock/ConditionBlock.tsx [hunk 1]
```diff
@@ -27,17 +27,50 @@ import ConditionBuilderAdd from '../ConditionBuilderAdd/ConditionBuilderAdd';
...
 
 /**
  * This component build each block of condition consisting of property, operator value and close button.
  */
 
-const ConditionBlock = (props) => {
-  const { property, value, operator } = props.condition;
+interface ConditionBlockProps {
+  condition?: Condition;
+  onRemove: (e: any) => void;
```
[the remark this answers is not in the document]
Comment (amal-k-joy) [reply packages/ibm-products/src/components/ConditionBuilder/ConditionBlock/ConditionBlock.tsx]: Hey,
I have made the changes

File: packages/ibm-products/src/components/ConditionBuilder/ConditionBuilderButton/ConditionBuilderButton.tsx [hunk 2]
```diff
@@ -11,9 +11,29 @@ import cx from 'classnames';
...
+  className?: string;
+  label: string;
+  hideLabel?: boolean;
+  tooltipAlign?: string;
+  renderIcon?: CarbonIconType;
+  onClick?: (e: any) => void;
+  onBlur?: React.FocusEventHandler<HTMLButtonElement>;
+  onFocus?: React.FocusEventHandler<HTMLButtonElement>;
+  //onBlur:any,
+  //onFocus:any,
```
[the remark this answers is not in the document]
Comment (amal-k-joy) [reply packages/ibm-products/src/components/ConditionBuilder/ConditionBuilderButton/ConditionBuilderButton.tsx]: Hey,
I have made the changes

File: packages/ibm-products/src/components/ConditionBuilder/ConditionBuilderButton/ConditionBuilderButton.tsx [hunk 3]
```diff
@@ -11,9 +11,29 @@ import cx from 'classnames';
...
+import { CarbonIconType, WarningAltFilled } from '@carbon/react/icons';
 import { usePrefix } from '@carbon/react';
 
+interface ConditionBuilderButtonProps {
+  className?: string;
+  label: string;
+  hideLabel?: boolean;
+  tooltipAlign?: string;
+  renderIcon?: CarbonIconType;
+  onClick?: (e: any) => void;
```
[the remark this answers is not in the document]
Comment (amal-k-joy) [reply packages/ibm-products/src/components/ConditionBuilder/ConditionBuilderButton/ConditionBuilderButton.tsx]: Hey,
I have made the changes

**Review by AlexanderMelox — APPROVED** [review]

**Review by davidmenendez — APPROVED** [review]

**Pull request merged**